pymeasure/display/widgets/table_widget.py

```diff
@@ -0,0 +1,713 @@
+    - Series length can be different, we call this number l(x), where x=1..n
+
+    The data can be presented as follow:
+    - By column: each series in a separate column, in this case table shape
+    will be: (k*n) x (max(l(x) x=1..n)
+    - By row: column fixed to the number of series, in this case table shape
+    will be: k x (sum of l(x) x=1..n)
+    """
+
+    float_digits = 6
+    concat_axis = 0
+
+    def __init__(self, column_index=None, results_list=[], parent=None):
+        super().__init__(parent)
+        self.column_index = column_index
+        self._init_data(results_list)
+
+    def _init_data(self, results_list=None):
+        if results_list is None:
+            results_list = []
+        self.results_list = results_list
+        self.row_count = self.pandas_row_count()
+        self.column_count = self.pandas_column_count()
+
+    def clear(self):
+        self.beginResetModel()
+        for results in self.results_list:
+            results.stop()
+        self._init_data()
+        self.endResetModel()
+
+    def add_results(self, results):
+        if results not in self.results_list:
+            self.beginResetModel()
+            self.results_list.append(results)
+            results.data_changed.connect(partial(self._data_changed, results))
+            self.endResetModel()
+            results.init()
+            results.start()
+            results.update_data()
+
+    def remove_results(self, results):
+        self.beginResetModel()
+        if results in self.results_list:
+            self.results_list.remove(results)
+        self.row_count = self.pandas_row_count()
+        self.column_count = self.pandas_column_count()
+        results.stop()
+        self.endResetModel()
+
+    def rowCount(self, parent=None):
+        return self.row_count
+
+    def columnCount(self, parent=None):
+        return self.column_count
+
+    def data(self, index, role=QtCore.Qt.ItemDataRole.DisplayRole):
+        if index.isValid() and role in (QtCore.Qt.ItemDataRole.DisplayRole, SORT_ROLE):
+            results, row, col = self.translate_to_local(index.row(), index.column())
+            try:
+                if (self.column_index is not None) and isinstance(self, PandasModelByColumn):
+                    # Remap row to matching index entry when layout is by column
+                    index = self.vertical_header[row]
+                    row = list(results.data.index).index(index)
+                value = results.data.iloc[row][col]
+                column_type = results.data.dtypes[col]
+                # Cast to column type
+                value_render = column_type.type(value)
+            except (IndexError, ValueError):
+                value = NaN
+                value_render = ""
+            if isinstance(value_render, float64):
+                # limit maximum number of decimal digits displayed
+                value_render = f"{value_render:.{self.float_digits:d}f}"
```

Comment:
the speficier `f` works fine for small numbers, but maybe to also support cases with large numbers, it would be better to use the `g` specifier?

Reply:
Yes, I think it is a good idea. I have done it.